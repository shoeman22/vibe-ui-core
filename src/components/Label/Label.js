import React from "react";
import PropTypes from "prop-types";

/** Label with required field display, htmlFor, and block styling and dot optionsssssddddd */
function Label({ htmlFor, label, required, requiredDotColor }) {
  return (
    <label style={{ display: "block" }} htmlFor={htmlFor}>
      {label}
      {required && <span style={{ color: requiredDotColor }}> *</span>}
    </label>
  );
}

Label.propTypes = {
  /** HTML ID for associated input */
  htmlFor: PropTypes.string.isRequired,

  /** Label text */
  label: PropTypes.string.isRequired,

  /** Display asterisk after label if true */
  required: PropTypes.bool,

  /** Color for required dot */
  requiredDotColor: PropTypes.string
};

Label.defaultProps = {
  required: false,
  requiredDotColor: "red"
};

export default Label;
