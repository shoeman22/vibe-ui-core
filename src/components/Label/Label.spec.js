import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Label from "./Label";

describe("Tests for Label component", () => {
  test("Check snapshot", () => {
    const tree = renderer
      .create(<Label htmlFor="testInput" label="Test Label" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Label matches expected text", () => {
    const label = shallow(<Label htmlFor="testInput" label="Test Label" />);
    expect(label.text()).toEqual("Test Label");
  });

  test("Required label has star", () => {
    const label = shallow(
      <Label htmlFor="testInput" label="Test Label" required />
    );
    expect(label.text()).toEqual("Test Label *");
  });
});
