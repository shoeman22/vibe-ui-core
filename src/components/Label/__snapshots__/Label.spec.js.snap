// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Label component Check snapshot 1`] = `
<label
  htmlFor="testInput"
  style={
    Object {
      "display": "block",
    }
  }
>
  Test Label
</label>
`;
